src/pbrt/util/taggedptr.h

```diff
@@ -825,10 +836,42 @@ class TaggedPointer {
     bool operator!=(const TaggedPointer &tp) const { return bits != tp.bits; }
 
     PBRT_CPU_GPU
-    void *ptr() { return reinterpret_cast<void *>(bits & ptrMask); }
+    void *ptr() {
+#if defined(PBRT_IS_WINDOWS) && defined(__HIPCC__)
+        unsigned int aptr = (bits >> tagShift) & 3;
```

Comment:
Since this has gotten complicated with the workaround, how about not duplicating the code but changing the non-const `ptr()` method to just be something like `return const_cast<void *>(((const TaggedPointer *)this)->ptr();` (Or an equivalent cleanup so that the code isn't duplicated.)